### ISSUE TITLE
false positive for not used when java is in project

### ISSUE THREAD

**Issue by edofic** [HairyFotr/linter#1]
Linter gives a false positive for "method parameter not used" if there's a java file in project.

full file http://pastebin.com/uRZcR0e5

it says this method doesn't use it's parameters

```
public void setAcceptedAttributeNameRegex(String acceptedAttributeNameRegex) {
    this.acceptedAttributeNameRegex = Pattern.compile(acceptedAttributeNameRegex);
}
```

and also this one

```
private boolean isAcceptedAttribute(Object key) {
    if (acceptedAttributeNameRegex == null) {
        return true;
    }
    if (!(key instanceof String)) {
        return true;
    }
    return acceptedAttributeNameRegex.matcher((String)key).matches();
}
```

since this is clearly false...is there a way to check if file being compiled is a java source and then just skip it(no linting)?
